doc/nrf/libraries/modem/location.rst

```diff
@@ -0,0 +1,198 @@
+      * Location library has implementation for WiFi location services
+      * Service is selected in :c:struct:`loc_method_config` when requesting location
+      * Data transport for the service is REST
+      * Available services are `nRF Cloud Location Services`_, `HERE Positioning`_ and `Skyhook Precision Location`_
+
+Supported features
+==================
+
+TODO: Not really sure what to put into this mandatory section given Implementation and Configuration sections has a lot of information and if something is missing, that can be added.
+
+.. note::
+   Use this section to describe the features supported by the library.
+
+Requirements
+************
+
+nRF Cloud certificates
+======================
+
+If you use nRF Cloud for any location data, you must have the certificate provisioned.
+See `Updating the nRF Connect for Cloud certificate`_ for more information.
+
+Location service accounts
+=========================
+
+To use the location services that provide A-GPS/P-GPS, cellular positioning and WiFi positioning data, see the respective documentation for account setup and for getting the required credentials for authentication:
+
+* `nRF Cloud Location Services`_
+* `HERE Positioning`_
+* `Skyhook Precision Location`_
+* `Polte Location API`_
+
+The required credentials for the location services are configurable using Kconfig options.
+
+WiFi chip
+=========
+
+WiFi is not supported by the HW of the supported DKs. External WiFi chips can be used such as ESP32, which you can connect to the DKs.
+
+TODO: any links we should add?
+
+Library files
+*************
+
+.. |library path| replace:: :file:`lib/location`
+
+This library is found under |library path| in the |NCS| folder structure.
+
+Configuration
+*************
+
+Configure the following Kconfig option to enable this library:
+
+* :kconfig:`CONFIG_LOCATION` - Enables the Location library.
+
+Configure the following options to enable location methods of your choice:
+
+* :kconfig:`CONFIG_LOCATION_METHOD_GNSS` - Enables GNSS location method.
+* :kconfig:`CONFIG_LOCATION_METHOD_CELLULAR` - Enables cellular location method.
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI` - Enables WiFi location method.
+* :kconfig:`CONFIG_WIFI` - Enable WiFi for Zephyr.
+
+The following options control the use of GNSS assistance data:
+
+* :kconfig:`CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL` - Enables A-GPS data retrieval from an external source which the application implements separately.
+* :kconfig:`CONFIG_NRF_CLOUD_AGPS` - Enables A-GPS data retrieval from `nRF Cloud`_.
+* :kconfig:`CONFIG_NRF_CLOUD_PGPS` - Enables P-GPS data retrieval from `nRF Cloud`_.
+
+The following options control the transport used with `nRF Cloud`_:
+
+* :kconfig:`CONFIG_NRF_CLOUD_REST` - Uses REST APIs to communicate with `nRF Cloud`_.
+* :kconfig:`CONFIG_NRF_CLOUD_MQTT` - Uses MQTT transport to communicate with `nRF Cloud`_.
+* :kconfig:`CONFIG_REST_CLIENT` - Enable :ref:`lib_rest_client` library.
```

Comment:
This is not currently used for GNSS method. Should we clarify here what is this good for? Similarly, can MQTT be used for all methods (cellular, wifi?)

Reply:
Implementation section comments about where REST and MQTT can be used so add a comment there if something is not true.

Reply:
@hiltunent Please close this or add further questions.